.github/workflows/test-all.yml

```
@@ -201,13 +204,69 @@ jobs:
           fi
 
       - name: "Run tests"
+        id: tests-default
         timeout-minutes: 12
         run: |
           PKG=${{ matrix.package }}
           gap -A --quitonbreak -r -c "SetInfoLevel(InfoPackageLoading, PACKAGE_WARNING);res:=TestPackage(\"$PKG\");FORCE_QUIT_GAP(res);"
 
       - name: "Run tests with OnlyNeeded"
+        id: tests-only-needed
+        if: ${{ always() }}
         timeout-minutes: 12
         run: |
           PKG=${{ matrix.package }}
           gap -A --quitonbreak -r -c "SetInfoLevel(InfoPackageLoading, PACKAGE_WARNING);res:=TestPackage(\"$PKG\":OnlyNeeded);FORCE_QUIT_GAP(res);"
+
+      # https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context
+      # steps.<id>.outcome possible values are success, failure, cancelled, or skipped.
+      - name: "Create job status json-file"
+        if: ${{ always() }}
+        run: |
+          PKG="${{ matrix.package }}"
+          STATUS_DEFAULT="${{ steps.tests-default.outcome }}"
+          STATUS_ONLY_NEEDED="${{ steps.tests-only-needed.outcome }}"
+          echo "{\"status_default\": \"$STATUS_DEFAULT\", \"status_only_needed\": \"$STATUS_ONLY_NEEDED\"}" > ${PKG}.json
+          cat ${PKG}.json
+
+      - name: "Upload job status as artifact"
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: "reports-${{ matrix.package }}"
+          path: "${{ matrix.package }}.json"
+
+  report:
+    name: "Report"
+    needs: test-package
+    if: ${{ always() }}
+    runs-on: ubuntu-18.04
```

Do we need to use an old Ubuntu for some reason? Then we should say so in a comment. Otherwise:
```suggestion
    runs-on: ubuntu-latest
```

---

This is probably copy-paste-pasta from some other old actions we used.